### ISSUE TITLE
Extension to country codes in folder and file-naming

### ISSUE THREAD

**Issue by TomazErjavec** [clarin-eric/ParlaMint#144]
So far, a ParlaMint corpus (file)name could have an ISO 3166 country code (such as GB),  and now this will need to be extended also to codes for country subdivisions (such as ES-PV for the Basque autonomous community), which probably impacts also on various scripts. 

However, we should probably also allow for a further extension giving the ISO 639 language code used. I don't think we should introduce this for the default cases (all that we have so far), but there are two situations where this would come in handy:
- we plan to MT all the corpora to English - the langauge name would distinguish them from the originals
- I was told that the Basque transcriptions are both in Basque and translated to Spanish,  but, if I remenber correctly, not everything is avalialbe in both languages, so we might have two sets of partially overlapping files. So, again a language code suffix could help to distinguish between the two.

In short, I propose that a legal ParlaMint filename / ID has the general form: 
`ParlaMint-<country>-<langauge>_<date>-<suffix>` or, more formally,
`ParlaMint-<ISO3166>-<ISO639>_<ISO8601>-<suffix>`, with `-<ISO639>`  and `-<suffix>` of course being optional. Note, however, that, in general, the ISO 639 language code can have three letters (although we will most likely always have two).

If this makes sense(?), this might also be the right time to enable the scripts to deal with it.

**Comment by TomazErjavec** [comment]
>     * We should be able to link translated version to original source version

OK, to an extent, i.e. I don't think we need to complicate to have e.g. `s/@corresp` for sentence alignment. I think it would be sufficient to have either the same IDs on elements that are alignmed or have a table giving the alignments. Not sure if that is what you were thinking about - because just indicating that two corpus directories contains the corpus in two languages kindof follows from the fact that one has the name ParlaMint-XX and the other ParlaMint-XX-yy. And, of course, we can put this info in the TEI header as well.

>     * Some source parliament can use multiple languages (I do not mean post-processed translation - I mean what has been said in parliament)

Of course. Like BE.
> 
>     * we probably want to distinguish the source version from the translations

But we will, via the language suffix. And, again, we should document this in the header.
 
> So I am suggesting:
> 
>     * The **source version** will use `ParlaMint-<country>` names. We don't care in the name how many languages has been used.

Yes.

>     * The **translated version** will use `ParlaMint-<country>-<langauge>`. There will be only one `<language>` in the text data and if someones want to determine the source version of data one should remove just `-<language>` from id.

Exactly.

> Another question, that crosses my mind is if there can be multiple translations to the same language:
>     * official and MT
>     * usage of different MT translators, ie for comparison of MT quality on the Parliament domain

This might be overthinking the issue, I am just worried about the official corpus releases. And, if we have an official translations, then we don't need MT. And if we have different MT versions, we will determine which one is best, and them pack that one into an official ParlaMint (MTed) corpus. Just for comparison, some ad-hoc encoding and IDs can be used. 

I am leaving the issue open, if nothing else, this filename extension still needs to be implemented.

**Comment by matyaskopp** [comment]
OK, I will take it into consideration.
 
> ... I think it would be sufficient to have either the same IDs on elements that are alignmed or have a table giving the alignments. ...

We can discuss this later, probably in a separate issue. We can make linking with `linkGrp/link` in a separate file(s), then we will need different ids. But currently, I don't have a definite opinion on this.

**Comment by TomazErjavec** [comment]
> OK, I will take it into consideration.
OK, thanks!

> > ... I think it would be sufficient to have either the same IDs on elements that are alignmed or have a table giving the alignments. ...
> 
> We can discuss this later, probably in a separate issue. We can make linking with `linkGrp/link` in a separate file(s), then we will need different ids. But currently, I don't have a definite opinion on this.

I agree, let's leave this for later, when it becomes important. And, in a new issue of course!
